matrix/matrix.py

```diff
@@ -0,0 +1,189 @@
+import operator
+from itertools import repeat
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type. This
+    implementation only supports even-side matrices (n * n).
```

Comment:
First row in doc should have short explanation about it.
It should be exactly 1 row (short and descriptive).
If you make something complex its ok to add notes in the doc below to explain it